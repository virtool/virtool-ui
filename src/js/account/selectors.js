import { get } from "lodash-es";

export const getAccount = state => state.account;
export const getAccountAdministratorRole = state => get(state, "account.administrator_role", null);

export const getAccountId = state => state.account.id;
export const getAccountHandle = state => state.account.handle;
