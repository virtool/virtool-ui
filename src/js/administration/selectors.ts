import { get } from "lodash-es";

export const getAccountAdministratorRole = state => get(state, "account.administrator_role", null);
