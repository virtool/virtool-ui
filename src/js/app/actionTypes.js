/**
 * Create a special action type used for requests.
 *
 * The request is an object with three properties assigned with action types used for API requests.
 *
 * - REQUESTED
 * - SUCCEEDED
 * - FAILED
 *
 * @param root {string} the root name of the action type
 * @returns {object} a request-style action type
 */
const createRequestActionType = root => ({
    REQUESTED: `${root}_REQUESTED`,
    SUCCEEDED: `${root}_SUCCEEDED`,
    FAILED: `${root}_FAILED`,
});

// App
export const GET_INITIAL_STATE = createRequestActionType("GET_INITIAL_STATE");

// Account
export const GET_ACCOUNT = createRequestActionType("GET_ACCOUNT");
export const UPDATE_ACCOUNT = createRequestActionType("UPDATE_ACCOUNT");
export const CHANGE_ACCOUNT_PASSWORD = createRequestActionType("CHANGE_ACCOUNT_PASSWORD");
export const LOGIN = createRequestActionType("LOGIN");
export const LOGOUT = createRequestActionType("LOGOUT");
export const RESET_PASSWORD = createRequestActionType("RESET_PASSWORD");

// Administration Settings
export const GET_SETTINGS = createRequestActionType("GET_SETTINGS");
export const UPDATE_SETTINGS = createRequestActionType("UPDATE_SETTINGS");

// Analysis
export const WS_UPDATE_ANALYSIS = "WS_UPDATE_ANALYSIS";
export const SET_ANALYSIS = "SET_ANALYSIS";
export const SET_ANALYSIS_ACTIVE_ID = "SET_ANALYSIS_ACTIVE_ID";
export const SET_SEARCH_IDS = "SET_SEARCH_IDS";
export const SET_AODP_FILTER = "ADD_AODP_FILTER";
export const SET_ANALYSIS_SORT_KEY = "SET_ANALYSIS_SORT_KEY";

// Errors
export const CLEAR_ERROR = "CLEAR_ERROR";

// Files
export const UPLOAD = createRequestActionType("UPLOAD");
export const UPLOAD_SAMPLE_FILE = createRequestActionType("UPLOAD_SAMPLE_FILE");
export const UPLOAD_PROGRESS = "UPLOAD_PROGRESS";
export const UPLOAD_FAILED = "UPLOAD_FAILED";
export const REMOVE_UPLOAD = "REMOVE_UPLOAD";

// Forms
export const SET_PERSISTENT_FORM_STATE = "SET_PERSISTENT_FORM_STATE";
export const DELETE_PERSISTENT_FORM_STATE = "DELETE_PERSISTENT_FORM_STATE";
export const SET_REDUX_FORM_STATE = "SET_REDUX_FORM_STATE";

// Groups
export const CREATE_GROUP = createRequestActionType("CREATE_GROUP");

// HMMs
export const GET_HMM = createRequestActionType("GET_HMM");

// Indexes
export const GET_INDEX = createRequestActionType("GET_INDEX");
export const CREATE_INDEX = createRequestActionType("CREATE_INDEX");

// Jobs
export const GET_JOB = createRequestActionType("GET_JOB");

// OTU
export const GET_OTU = createRequestActionType("GET_OTU");
export const CREATE_OTU = createRequestActionType("CREATE_OTU");
export const EDIT_OTU = createRequestActionType("EDIT_OTU");
export const ADD_ISOLATE = createRequestActionType("ADD_ISOLATE");
export const EDIT_ISOLATE = createRequestActionType("EDIT_ISOLATE");
export const ADD_SEQUENCE = createRequestActionType("ADD_SEQUENCE");
export const EDIT_SEQUENCE = createRequestActionType("EDIT_SEQUENCE");

// Refs
export const GET_REFERENCE = createRequestActionType("GET_REFERENCE");

// Samples
export const GET_SAMPLE = createRequestActionType("GET_SAMPLE");
export const CREATE_SAMPLE = createRequestActionType("CREATE_SAMPLE");
export const UPDATE_SAMPLE = createRequestActionType("UPDATE_SAMPLE");

// Subtraction
export const GET_SUBTRACTION = createRequestActionType("GET_SUBTRACTION");
export const CREATE_SUBTRACTION = createRequestActionType("CREATE_SUBTRACTION");

// Users
export const FIND_USERS = createRequestActionType("FIND_USERS");
export const GET_USER = createRequestActionType("GET_USER");
export const CREATE_USER = createRequestActionType("CREATE_USER");
export const CREATE_FIRST_USER = createRequestActionType("CREATE_FIRST_USER");
export const EDIT_USER = createRequestActionType("EDIT_USER");
