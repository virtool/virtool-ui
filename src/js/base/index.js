export { Accordion } from "./accordion/Accordion";
export { AccordionContent } from "./accordion/AccordionContent";
export { AccordionTrigger } from "./accordion/AccordionTrigger";
export { ScrollingAccordionItem } from "./accordion/ScrollingAccordionItem";
export { Alert } from "./Alert";
export { AlertInner } from "./AlertInner";
export { AlertOuter } from "./AlertOuter";
export { Attribution } from "./Attribution";
export { AttributionWithName } from "./AttributionWithName";
export { Badge } from "./Badge";
export { Box } from "./Box";
export { BoxGroup } from "./BoxGroup";
export { BoxGroupDisabled } from "./BoxGroupDisabled";
export { BoxGroupHeader } from "./BoxGroupHeader";
export { BoxGroupSearch } from "./BoxGroupSearch";
export { BoxGroupSection } from "./BoxGroupSection";
export { BoxGroupTable } from "./BoxGroupTable";
export { BoxSpaced } from "./BoxSpaced";
export { BoxTitle } from "./BoxTitle";
export { device } from "./Breakpoints";
export { Button } from "./Button";
export { Checkbox } from "./Checkbox";
export { CheckboxLabel } from "./CheckboxLabel";
export { Circle } from "./Circle";
export { CloseButton } from "./CloseButton";
export { Color } from "./Color";
export { ComboBox } from "./combobox/ComboBox";
export { Container } from "./Container";
export { ContainerNarrow } from "./ContainerNarrow";
export { ContainerSide } from "./ContainerSide";
export { ContainerWide } from "./ContainerWide";
export { Dialog } from "./Dialog";
export { DialogAlert } from "./DialogAlert";
export { DialogContent } from "./DialogContent";
export { DialogContext } from "./DialogContext";
export { DialogFooter } from "./DialogFooter";
export { DialogOverlay } from "./DialogOverlay";
export { DialogTabs } from "./DialogTabs";
export { DialogTitle } from "./DialogTitle";
export { DialogTrigger } from "./DialogTrigger";
export { DividerHorizontal } from "./DividerHorizontal";
export { DividerVertical } from "./DividerVertical";
export { Dropdown } from "./Dropdown";
export { DropdownButton } from "./DropdownButton";
export { DropdownMenuContent } from "./DropdownMenuContent";
export { DropdownMenuDownload } from "./DropdownMenuDownload";
export { DropdownMenuItem } from "./DropdownMenuItem";
export { DropdownMenuLink } from "./DropdownMenuLink";
export { ExternalLink } from "./ExternalLink";
export { Icon } from "./Icon";
export { IconButton } from "./IconButton";
export { InitialIcon } from "./InitialIcon";
export { Input } from "./Input";
export { InputContainer } from "./InputContainer";
export { InputContext } from "./InputContext";
export { InputError } from "./InputError";
export { InputGroup } from "./InputGroup";
export { InputIcon } from "./InputIcon";
export { InputIconButton } from "./InputIconButton";
export { InputLabel } from "./InputLabel";
export { InputLevel } from "./InputLevel";
export { InputLoading } from "./InputLoading";
export { InputPassword } from "./InputPassword";
export { InputSearch } from "./InputSearch";
export { InputSelect } from "./InputSelect";
export { InputSimple } from "./InputSimple";
export { Key } from "./Key";
export { Label } from "./Label";
export { LinkButton } from "./LinkButton";
export { Loader } from "./Loader";
export { LoadingPlaceholder } from "./LoadingPlaceholder";
export { Logo } from "./Logo";
export { Markdown } from "./Markdown";
export { NoneFound } from "./NoneFound";
export { NoneFoundBox } from "./NoneFoundBox";
export { NoneFoundSection } from "./NoneFoundSection";
export { NotFound } from "./NotFound";
export { Pagination } from "./pagination/Pagination";
export { ProgressBar } from "./ProgressBar";
export { ProgressBarAffixed } from "./ProgressBarAffixed";
export { ProgressCircle } from "./ProgressCircle";
export { RelativeTime } from "./RelativeTime";
export { RemoveBanner } from "./RemoveBanner";
export { SaveButton } from "./SaveButton";
export { SectionHeader } from "./SectionHeader";
export { Select } from "./select/Select";
export { SelectButton } from "./select/SelectButton";
export { SelectContent } from "./select/SelectContent";
export { SelectItem } from "./select/SelectItem";
export { SelectBox } from "./SelectBox";
export { SelectBoxGroupSection } from "./SelectBoxGroupSection";
export { SidebarHeader } from "./SidebarHeader";
export { SidebarHeaderButton } from "./SidebarHeaderButton";
export { SideBarSection } from "./SideBarSection";
export { SlashList } from "./SlashList";
export { SubviewHeader } from "./SubviewHeader";
export { SubviewHeaderAttribution } from "./SubviewHeaderAttribution";
export { SubviewHeaderTitle } from "./SubviewHeaderTitle";
export { Table } from "./Table";
export { Tabs } from "./Tabs";
export { TabsLink } from "./TabsLink";
export { TextArea } from "./TextArea";
export { Toolbar } from "./Toolbar";
export { ViewHeader } from "./ViewHeader";
export { ViewHeaderAttribution } from "./ViewHeaderAttribution";
export { ViewHeaderIcons } from "./ViewHeaderIcons";
export { ViewHeaderSubtitle } from "./ViewHeaderSubtitle";
export { ViewHeaderTitle } from "./ViewHeaderTitle";
