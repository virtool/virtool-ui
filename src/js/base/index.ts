import InputGroup from "@base/InputGroup.js";
import InputLabel from "@base/InputLabel.js";

export { InputGroup, InputLabel };

export { Link } from "wouter";
export { Checkbox } from "./Checkbox";
export { CloseButton } from "./CloseButton";
export { Color } from "./Color";
export { ComboBox } from "./ComboBox";
export { Container } from "./Container";
export { ContainerNarrow } from "./ContainerNarrow";
export { ContainerSide } from "./ContainerSide";
export { ContainerWide } from "./ContainerWide";
export { Dialog } from "./Dialog";
export { DialogAlert } from "./DialogAlert";
export { DialogContent } from "./DialogContent";
export { DialogContext } from "./DialogContext";
export { DialogFooter } from "./DialogFooter";
export { DialogOverlay } from "./DialogOverlay";
export { DialogTabs } from "./DialogTabs";
export { DialogTitle } from "./DialogTitle";
export { DialogTrigger } from "./DialogTrigger";
export { Dropdown } from "./Dropdown";
export { DropdownButton } from "./DropdownButton";
export { DropdownMenuContent } from "./DropdownMenuContent";
export { DropdownMenuDownload } from "./DropdownMenuDownload";
export { DropdownMenuItem } from "./DropdownMenuItem";
export { DropdownMenuLink } from "./DropdownMenuLink";
export { ExternalLink } from "./ExternalLink";
export { Icon } from "./Icon";
export { IconButton } from "./IconButton";
export { InitialIcon } from "./InitialIcon";
export { Input } from "./Input";
export { InputContainer } from "./InputContainer";
export { InputContext } from "./InputContext";
export { InputError } from "./InputError";
export { InputIcon } from "./InputIcon";
export { InputIconButton } from "./InputIconButton";
export { InputLoading } from "./InputLoading";
export { InputPassword } from "./InputPassword";
export { InputSearch } from "./InputSearch";
export { InputSelect } from "./InputSelect";
export { InputSimple } from "./InputSimple";
export { Key } from "./Key";
export { Label } from "./Label";
export { LinkButton } from "./LinkButton";
export { Loader } from "./Loader";
export { LoadingPlaceholder } from "./LoadingPlaceholder";
export { Logo } from "./Logo";
export { Markdown } from "./Markdown";
export { NoneFound } from "./NoneFound";
export { NoneFoundBox } from "./NoneFoundBox";
export { NoneFoundSection } from "./NoneFoundSection";
export { NotFound } from "./NotFound";
export { Pagination } from "./Pagination";
export { ProgressBar } from "./ProgressBar";
export { ProgressBarAffixed } from "./ProgressBarAffixed";
export { ProgressCircle } from "./ProgressCircle";
export { RelativeTime } from "./RelativeTime";
export { RemoveBanner } from "./RemoveBanner";
export { SaveButton } from "./SaveButton";
export { SectionHeader } from "./SectionHeader";
export { Select } from "./Select";
export { SelectBox } from "./SelectBox";
export { SelectBoxGroupSection } from "./SelectBoxGroupSection";
export { SelectButton } from "./SelectButton";
export { SelectContent } from "./SelectContent";
export { SelectItem } from "./SelectItem";
export { SidebarHeader } from "./SidebarHeader";
export { SidebarHeaderButton } from "./SidebarHeaderButton";
export { SideBarSection } from "./SideBarSection";
export { SlashList } from "./SlashList";
export { SubviewHeader } from "./SubviewHeader";
export { SubviewHeaderAttribution } from "./SubviewHeaderAttribution";
export { SubviewHeaderTitle } from "./SubviewHeaderTitle";
export { Table } from "./Table";
export { Tabs } from "./Tabs";
export { TabsLink } from "./TabsLink";
export { TextArea } from "./TextArea";
export { ViewHeader } from "./ViewHeader";
export { ViewHeaderAttribution } from "./ViewHeaderAttribution";
export { ViewHeaderIcons } from "./ViewHeaderIcons";
export { ViewHeaderSubtitle } from "./ViewHeaderSubtitle";
export { ViewHeaderTitle } from "./ViewHeaderTitle";
