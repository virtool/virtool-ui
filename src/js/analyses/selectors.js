import { filter, find, intersection, keyBy, map, reject, sortBy, toNumber, toString } from "lodash-es";
import { createSelector } from "reselect";
import { getMaxReadLength } from "../samples/selectors";
import { createFuse } from "../utils/utils";
import { fuseSearchKeys } from "./utils";

export const getActiveId = state => state.analyses.activeId;
export const getFilterOTUs = state => state.analyses.filterOTUs;
export const getFilterSequences = state => state.analyses.filterSequences;
export const getFilterAODP = state => state.analyses.filterAODP;
export const getHits = state => state.analyses.detail.results.hits;
export const getReadCount = state => state.analyses.detail.results.readCount;
export const getResults = state => state.analyses.detail.results;
export const getWorkflow = state => state.analyses.detail.workflow;

/**
 * Return a Fuse object for searching through results given an workflow name. The workflow will determine which keys the
 * search runs over.
 *
 */
export const getFuse = createSelector([getWorkflow, getHits], (workflow, hits) =>
    createFuse(hits, fuseSearchKeys[workflow], "id")
);

export const getFilterIds = createSelector(
    [getFilterAODP, getWorkflow, getHits, getFilterOTUs, getFilterSequences, getMaxReadLength, getReadCount],
    (aodpFilter, workflow, hits, filterOTUs, filterSequences, maxReadLength, readCount) => {
        if (workflow === "nuvs") {
            const filteredHits = filterSequences ? reject(hits, { e: undefined }) : hits;
            return map(filteredHits, "id");
        }

        if (workflow === "pathoscope_bowtie" && filterOTUs) {
            const filteredHits = reject(hits, hit => {
                return hit.pi * readCount < (hit.length * 0.8) / maxReadLength;
            });

            return map(filteredHits, "id");
        }

        if (workflow === "aodp" && aodpFilter) {
            const filteredHits = filter(hits, hit => {
                if (hit.identity > aodpFilter * 100) {
                    return hit.id;
                }
            });

            return map(filteredHits, fi => fi.id);
        }

        return map(hits, "id");
    }
);

export const getSearchIds = state => state.analyses.searchIds;

export const getSortKey = state => state.analyses.sortKey;

export const getSortIds = createSelector([getWorkflow, getHits, getSortKey], (workflow, hits, sortKey) => {
    switch (sortKey) {
        case "e":
            return map(sortBy(hits, "e"), "id");

        case "orfs":
            return map(sortBy(hits, "annotatedOrfCount").reverse(), "id");

        case "length":
            return map(sortBy(hits, "sequence.length").reverse(), "id");

        case "depth":
            return map(sortBy(hits, "depth").reverse(), "id");

        case "coverage":
            return map(sortBy(hits, "coverage").reverse(), "id");

        case "weight":
            return map(sortBy(hits, "pi").reverse(), "id");

        case "identity":
            return map(sortBy(hits, "identity").reverse(), "id");

        default:
            return map(hits, "id");
    }
});

export const getMatches = createSelector(
    [getWorkflow, getHits, getFilterIds, getSearchIds, getSortIds],
    (workflow, hits, filterIds, searchIds, sortIds) => {
        const matchIds = searchIds
            ? intersection(sortIds, filterIds, map(searchIds, workflow === "nuvs" ? toNumber : toString))
            : intersection(sortIds, filterIds);

        const keyed = keyBy(hits, "id");

        return map(matchIds, id => keyed[id]);
    }
);

export const getActiveHit = createSelector([getWorkflow, getMatches, getActiveId], (workflow, matches, activeId) => {
    if (activeId !== null) {
        const hit = find(matches, { id: activeId });

        if (hit) {
            return hit;
        }
    }

    return matches[0] || null;
});
