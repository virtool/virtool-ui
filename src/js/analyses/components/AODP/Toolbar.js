import { toNumber } from "lodash-es";
import React, { useCallback } from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { Input, SearchInput, Toolbar } from "../../../base";
import { setAnalysisSortKey, setAODPFilter, setSearchIds } from "../../actions";
import { getFuse } from "../../selectors";
import { AnalysisViewerSort } from "../Viewer/Sort";

const StyledAODPToolbar = styled(Toolbar)`
    input[type="number"] {
        max-width: 130px;
    }
`;

export const AODPToolBar = ({ filterAODP, fuse, id, sortKey, onSearch, onSelect, onSetFilter }) => {
    const handleChangeSearch = useCallback(
        e => {
            onSearch(e.target.value, fuse);
        },
        [id]
    );

    const handleChangeFilter = e => {
        onSetFilter(toNumber(e.target.value));
    };

    return (
        <StyledAODPToolbar>
            <SearchInput onChange={handleChangeSearch} onKeyDown={e => e.stopPropagation()} placeholder="Name" />
            <AnalysisViewerSort workflow="aodp" sortKey={sortKey} onSelect={onSelect} />
            <Input
                type="number"
                placeholder="Minimum Identity"
                min={0.5}
                max={1.0}
                step={0.01}
                value={filterAODP.toFixed(2)}
                onChange={e => handleChangeFilter(e)}
            />
        </StyledAODPToolbar>
    );
};

const mapStateToProps = state => ({
    filterAODP: state.analyses.filterAODP,
    fuse: getFuse(state),
    id: state.analyses.detail.id,
    sortKey: state.analyses.sortKey
});

const mapDispatchToProps = dispatch => ({
    onSearch: (term, fuse) => {
        dispatch(setSearchIds(term ? fuse.search(term) : null));
    },

    onSelect: sortKey => {
        dispatch(setAnalysisSortKey(sortKey));
    },

    onSetFilter: identity => {
        dispatch(setAODPFilter(identity));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(AODPToolBar);
