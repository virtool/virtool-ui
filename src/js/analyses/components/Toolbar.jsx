import React from "react";
import { connect } from "react-redux";
import { pushState } from "../../app/actions";
import { Button, InputSearch, Toolbar } from "../../base";
import { getCanModify } from "../../samples/selectors";
import { findAnalyses } from "../actions";

export function AnalysesToolbar({ canModify, onFind, onShowCreate, page, sampleId, term }) {
    return (
        <Toolbar>
            <InputSearch value={term} onChange={e => onFind(sampleId, e.target.value, page)} />
            <Button
                icon="plus-square fa-fw"
                tip="New Analysis"
                color="blue"
                onClick={() => onShowCreate(sampleId)}
                disabled={!canModify}
            />
        </Toolbar>
    );
}

export function mapStateToProps(state) {
    return {
        canModify: getCanModify(state),
        page: state.analyses.page,
        sampleId: state.samples.detail.id,
        term: state.analyses.term,
    };
}

export function mapDispatchToProps(dispatch) {
    return {
        onFind: (sampleId, term, page) => {
            dispatch(findAnalyses(sampleId, term, page));
        },
        onShowCreate: sampleId => {
            dispatch(pushState({ createAnalysis: sampleId }));
        },
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(AnalysesToolbar);
