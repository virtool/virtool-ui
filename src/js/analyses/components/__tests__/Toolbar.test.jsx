vi.mock("../../../samples/selectors");

import { FIND_ANALYSES, PUSH_STATE } from "../../../app/actionTypes";
import { Button, InputSearch } from "../../../base";
import { getCanModify } from "../../../samples/selectors";
import { AnalysesToolbar, mapDispatchToProps, mapStateToProps } from "../Toolbar";

describe("<AnalysesToolbar />", () => {
    let props;

    beforeEach(() => {
        props = {
            canModify: true,
            page: 2,
            sampleId: "foobar",
            term: "baz",
            onFind: vi.fn(),
            onShowCreate: vi.fn()
        };
    });

    it("should render with default props", () => {
        const wrapper = shallow(<AnalysesToolbar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render with [canModify=false]", () => {
        props.canModify = false;
        const wrapper = shallow(<AnalysesToolbar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should call onChange", () => {
        const wrapper = shallow(<AnalysesToolbar {...props} />);
        const event = {
            target: {
                value: "hello"
            }
        };
        wrapper.find(InputSearch).simulate("change", event);
        expect(props.onFind).toHaveBeenCalledWith(props.sampleId, "hello", props.page);
    });

    it("should call onShowCreate", () => {
        const wrapper = shallow(<AnalysesToolbar {...props} />);
        wrapper.find(Button).simulate("click");
        expect(props.onShowCreate).toHaveBeenCalledWith(props.sampleId);
    });
});

describe("mapStateToProps", () => {
    let state;
    let expected;

    beforeEach(() => {
        state = {
            analyses: {
                page: 2,
                term: "baz"
            },
            samples: {
                detail: {
                    id: "foo"
                }
            }
        };
        expected = {
            canModify: true,
            page: 2,
            sampleId: "foo",
            term: "baz"
        };
    });

    it("should return props when [canModify=true]", () => {
        getCanModify.mockReturnValue(true);
        const result = mapStateToProps(state);
        expect(result).toEqual(expected);
    });

    it("should return props when [canModify=false]", () => {
        getCanModify.mockReturnValue(false);
        const result = mapStateToProps(state);
        expected.canModify = false;
        expect(result).toEqual(expected);
    });
});

describe("mapDispatchToProps", () => {
    let dispatch;
    let props;

    beforeEach(() => {
        dispatch = vi.fn();
        props = mapDispatchToProps(dispatch);
    });

    it("should return onShowCreate in props", () => {
        const dispatch = vi.fn();
        const props = mapDispatchToProps(dispatch);

        // Call onShowCreate.
        props.onShowCreate("foobar");
        expect(dispatch).toHaveBeenCalledWith({
            type: PUSH_STATE,
            payload: {
                state: {
                    createAnalysis: "foobar"
                }
            }
        });
    });

    it("should return onFind in props", () => {
        // Call onFind.
        const sampleId = "foobar";
        const term = "baz";
        const page = 3;

        props.onFind(sampleId, term, page);
        expect(dispatch).toHaveBeenCalledWith({
            type: FIND_ANALYSES.REQUESTED,
            payload: { sampleId, term, page }
        });
    });
});
