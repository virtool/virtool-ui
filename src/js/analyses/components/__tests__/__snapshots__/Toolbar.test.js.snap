// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnalysesToolbar /> should render with [canModify=false] 1`] = `
<Toolbar>
  <SearchInput
    onChange={[Function]}
    value="baz"
  />
  <Button
    color="blue"
    disabled={true}
    icon="plus-square fa-fw"
    onClick={[Function]}
    tip="New Analysis"
    tipPlacement="top"
    type="button"
  />
</Toolbar>
`;

exports[`<AnalysesToolbar /> should render with default props 1`] = `
<Toolbar>
  <SearchInput
    onChange={[Function]}
    value="baz"
  />
  <Button
    color="blue"
    disabled={false}
    icon="plus-square fa-fw"
    onClick={[Function]}
    tip="New Analysis"
    tipPlacement="top"
    type="button"
  />
</Toolbar>
`;
